File: frontend/src/app/dtos/booking/reservation-purchase.dto.ts
import {PaymentInfoDto} from './payment-info.dto';

export class ReservationPurchaseDto {
  constructor(
    public seatingPlanId: number,
    public bookedSeats: Seat[],
    public bookedNonSeats: NonSeat[],
    public cost: number,
    public paymentInfoDto: PaymentInfoDto
  ) {}
}
  export class Seat {
  constructor(
    public seat: number,
    public row: number,
    public sector: number
  ) {}
}

export class NonSeat {
  constructor(
    public sector: number,
    public amount: number
  ) {}
}

File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/endpoint/dto/booking/ReservationPurchaseDto.java
package at.ac.tuwien.sepm.groupphase.backend.endpoint.dto.booking;

import java.math.BigDecimal;
import java.util.List;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;

public record ReservationPurchaseDto(
    @NotNull(message = "must not be null") Long seatingPlanId,
    List<@NotNull(message = "must not be null") Seat> bookedSeats,
    List<@NotNull(message = "must not be null") NonSeat> bookedNonSeats,
    @NotNull(message = "must not be null") BigDecimal cost,
    @NotNull(message = "must not be null") PaymentInfoDto paymentInfoDto) {

  public record Seat(@NotNull Long seat, @NotNull Long row, @NotNull Long sector) {}

  public record NonSeat(
      @NotNull Long sector,
      @Pattern(regexp = "^[1-9]+[0-9]*$", message = "must be a positive number") Integer amount) {}
}
